includes/class-wc-payments-webhook-processing-service.php

```diff
@@ -198,6 +200,30 @@ public function process( array $event_body ) {
 		}
 	}
 
+	/**
+	 * Check webhook mode against the gateway mode.
+	 *
+	 * @param array $event_body The event that triggered the webhook.
+	 *
+	 * @throws Invalid_Webhook_Data_Exception Event mode does not match the gateway mode.
+	 */
+	private function check_webhook_mode_mismatch( array $event_body ) {
+		$is_gateway_live_mode = ! $this->wcpay_gateway->is_in_test_mode();
+		$is_event_live_mode   = $this->read_webhook_property( $event_body, 'livemode' );
+
+		if ( $is_gateway_live_mode !== $is_event_live_mode ) {
+			$event_id = $this->read_webhook_property( $event_body, 'id' );
+
+			throw new Invalid_Webhook_Data_Exception(
```

Comment:
I think it might be worth just logging the mismatch, and bailing without throwing an exception. Switching from one mode to another, and receiving a webhook for the old mode is not such an exceptional case, and we probably don't need to raise any alarms in that case.

Reply:
It makes sense.
I slightly changed the code to log an error and bail in case there's a mismatch if you'd like to review it again.
Reg. unit tests, I didn't find better ways to verify the change than checking that the process for an easy event (here a notification) is not made in case of a mismatch (test event + live gateway or the opposite). It was easier to check with exceptions, but that should be enough here

Reply:
Thank you for switching to logging! The test is (and was) sufficient IMO, marking this as resolved.